GVRf/Framework/framework/src/main/java/org/gearvrf/animation/GVRSkeleton.java

```diff
@@ -90,12 +113,14 @@
     public static final int BONE_PHYSICS = 2;
 
     private static final String TAG = Log.tag(GVRSkeleton.class);
-    final protected GVRSceneObject[] mBones;
-    final protected int[] mParentBones;
-    final protected int[] mBoneOptions;
+    protected GVRSceneObject[] mBones;
+    protected int[] mParentBones;
+    protected int[] mBoneOptions;
     final private Quaternionf mTempQuatA = new Quaternionf();
     final private Quaternionf mTempQuatB = new Quaternionf();
     final private Matrix4f mTempMtx = new Matrix4f();
+    private static int[] sTempBoneParents;
+    private static GVRSceneObject sTempRoot;
```

Comment:
Static references to GVRf objects are not a good idea in general. In particular the scene object referenced will never be garbage collected. So it is a memory leak.

Reply:
sTempRoot still around. 

Can be changed to
```
        super(root.getGVRContext(), 0);
        setNative(NativeSkeleton.ctor(makeParentBoneIds(root, boneNames)));
```

Will allow makeParentBoneIds to be instance method and sTempRoot to be removed.